apps/visualisation/opencv_visualisation.cpp

```diff
@@ -85,7 +85,10 @@ int main( int argc, const char** argv )
         "{ help h usage ? |      | show this message }"
         "{ image i        |      | (required) path to reference image }"
         "{ model m        |      | (required) path to cascade xml file }"
-        "{ data d         |      | (optional) path to video output folder }"
+        "{ data d         |      | (optional) path to video output folder incl. trailing path separator }"
```

Comment:
"incl. trailing path separator" - it's error prone solution. Double slash is usually handled by OS and works well. The path without slash leads to invalid location and app failure.

Reply:
> ... The path without slash leads to invalid location and app failure.

Interesting. On my computer `--data=temp_data` instead of `--data=temp_data/` technically works i.e. files are created, not in the output folder but in the parent folder of the output folder.

```
temp_datamodel_visualization.mp4
temp_datastage_0.png
temp_datastage_1.png
...
```

Continuing that thought:
* if the parent folder was inaccessible due to permissions then the app would fail
* as a user due to the "path to video output folder" wording I would consider "files in the parent folder" as a failure

Either way though, changing the "data" usage message or the code logic w.r.t. missing trailing path separator is strictly speaking not in "configurable video codec" scope and so let me go and remove it accordingly then.

```suggestion
        "{ data d         |      | (optional) path to video output folder }"
```